crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -519,6 +525,55 @@ where
         }
     }
 
+    fn reverse_prefix_iter_keys(
+        &self,
+        prefix: &[u8],
+        column: Description::Column,
+    ) -> impl Iterator<Item = KeyItem> + '_ {
+        let maybe_next_item = next_prefix(prefix.to_vec())
+            .and_then(|next_prefix| {
+                self.iter_store_keys(
+                    column,
+                    Some(next_prefix.as_slice()),
+                    None,
+                    IterDirection::Forward,
+                )
+                .next()
+            })
+            .and_then(|res| res.ok());
+
+        let prefix_vec = prefix.to_vec(); // Convert `prefix` to an owned `Vec<u8>`
```

Comment:
This probably doesn't need to be `to_vec` you can just use the slice below.

Reply:
It looks like it will be painful trying to get the lifetimes to align here. It's probable that we can just use a constant sized array in the future, rather than a vec, in which case this shouldn't be a problem.

Not really a problem now either, but worth optimizing for sure.